packages/site/src/content/Button/Button.props.json

```diff
@@ -0,0 +1,4664 @@
+[
+  {
+    "tags": {},
+    "filePath": "../components/src/Button/Button.tsx",
+    "description": "",
+    "displayName": "Button",
+    "methods": [],
+    "props": {
+      "external": {
+        "defaultValue": null,
+        "description": "",
+        "name": "external",
+        "parent": {
+          "fileName": "../components/src/Button/Button.tsx",
+          "name": "ButtonFoundationProps"
+        },
+        "declarations": [
+          {
+            "fileName": "../components/src/Button/Button.tsx",
+            "name": "ButtonFoundationProps"
+          },
+          {
+            "fileName": "../components/src/Button/Button.tsx",
+            "name": "ButtonFoundationProps"
```

Comment:
Hokay. So!

I did the dive. I agree that it's gross. I saw it when I started with Button, but moved on because I figured it was a by-product of the generation and I had other problems to solve.

The fix? We need to fix our types.

It's the XOR statements in Button.tsx. They really throw react-docgen-typescript for a loop (My guess is this is already happening on the storybook side, we just don't see it because the generated props are a by-product of the storybook run and we don't see them)

What the declarations are, is every time the prop is declared in the file. If you remove all the XOR statements with singular `Partial` typescript usages (which does 'break' what XOR is doing, but only in the sense that it no longer disallows incorrect prop combinations), you drastically reduce the size of the Button.props.json file. In there it's clear that every duplication of 'ButtonFoundationProps' is because one of the other types in the Button has extended from it (I think nearly all of them do). So instead of 400 duplications, you have 7 duplications, one for every extended type.

When we chat today about the diff, let's chat about this too.